pkg/cloud/affinity_groups_test.go

```diff
@@ -48,22 +55,79 @@ var _ = Describe("AffinityGroup Unit Tests", func() {
 		mockCtrl.Finish()
 	})
 
-	It("fetches an affinity group", func() {
-		dummies.AffinityGroup.ID = "" // Force name fetching.
-		ags.EXPECT().GetAffinityGroupByName(dummies.AffinityGroup.Name).Return(&cloudstack.AffinityGroup{}, 1, nil)
+	Context("Fetch or Create Affinity group", func() {
```

Comment:
Contexts are generally situational. 

https://onsi.github.io/ginkgo/#organizing-specs-with-container-nodes

You may want to use a describe here instead of a context.

``` Go
  Describe("Extracting the author's first and last name", func() {
    Context("When the author has both names", func() {
      It("can extract the author's last name", func() {   
```

Of course they all do the same thing though. : )